File: Semester 5 Fall 2024/CL3001-Computer Networks - Lab/Lab Manuals/Lab Manual # 1/Python code/Q1PartD.py

def containsVowels(word):
    for ch in word:
        if(ch in ['a','e','i','o','u','A','E','I','O','U']):
            return True
    return False
        
def invert(word):
    e=len(word)-1
    reversedString=""
    while(e>=0):
        reversedString= reversedString + word[e]
        e=e-1
    return reversedString

def main():
    outputFile= open("PartD_Output.txt",'w')
    
    with open("Input File.txt",'r') as f:
        for line in f:
            words=line.split()
            for word in words:
                if(containsVowels(word)==True):
                    outputFile.write(invert(word))
                else:
                    outputFile.write(word)
                outputFile.write(" ")
            outputFile.write('\n')
        
    outputFile.close()





main()

File: Semester 5 Fall 2024/CL3001-Computer Networks - Lab/Lab Manuals/Lab Manual # 1/C code/Q1PartD.c
#include <stdio.h>
#include <string.h>

void invert(char * buff,int size)
{
	int s=0;
	int e=size-1;

	while(s<e)
	{
		char temp=buff[s];
		buff[s]=buff[e];
		buff[e]=temp;
		s=s+1;
		e=e-1;
	}
}

int main(int argc, char * argv[])
{
	if(argc!=2)
	{
		printf("Invalid parameters! Give one file name.\n");
		return 0;
	}

	FILE *read = fopen(argv[1], "r");
	if(read==NULL)
	{
		printf("Cannot open input file!\n");
		return 0;
	}

	FILE *write = fopen("PartD_Output.txt", "w"); 
	if(write==NULL)
	{
		printf("Cannot open output file!\n");
		return 0;
	}

	char buff[200];
	int status = fscanf(read, "%s", buff);
	
	while(status!=EOF)
	{
		int i = 0;
		int vowelFound = 0;

		for(i = 0; buff[i] != '\0'; i++)
		{
			if ((buff[i] == 'a') || (buff[i] == 'e') ||
			 (buff[i] == 'i') || (buff[i] == 'o') || (buff[i] == 'u') || 
			 (buff[i] == 'A') || (buff[i] == 'E') ||
			 (buff[i] == 'I') || (buff[i] == 'O') || (buff[i] == 'U'))
			 {
				vowelFound = 1;
			 }
		}
		
		if (vowelFound==1)
		{
			invert(buff,strlen(buff));
		}
		
		fputs(buff, write);
		fputc(' ', write);
		status= fscanf(read, "%s", buff);
	}
	
	printf("Inverted word file generated successfully! with the name: PartD_Output.txt\n");
	
	fclose(read);
	fclose(write);
	return 0;
}


